### ISSUE TITLE
Ignore events from certain sources

### ISSUE THREAD

**Issue by broyuken** [danobot/entity-controller#192]
Now that HA lists sources of changes in event logs, can we use this info to ignore changes from certain sources like itself (for that 2 second grace period) or from other apps like circadian lighting?

**Comment by danobot** [comment]
please provide some sample code

**Comment by broyuken** [comment]
https://www.home-assistant.io/blog/2020/09/17/release-115/#more-information-in-the-logbook-panel

I don’t have sample code, but I see that this is now available in the logs

**Comment by danobot** [comment]
Got it, please test on `issue/192` branch :)

**Comment by broyuken** [comment]
Downloading this now. If this works, could you also add the ability to add other sources to ignore? If this could also ignore circadian lighting this would be a perfect combo for my lighting.

**Comment by broyuken** [comment]
![B2700EE9-0250-4909-833C-F45A32739CD7](https://user-images.githubusercontent.com/1593391/93878661-6c4e5680-fca8-11ea-9ecc-0165592da95b.jpeg)

Did not work, it still blocked itself

**Comment by danobot** [comment]
Was this working before?

**Comment by broyuken** [comment]
This was working fine with the grace_period, but is not now. I will turn debug back on and send you the logs.

**Comment by broyuken** [comment]
Just downloaded your newest fix from like 30 minutes ago and now it seems to work.

**Comment by broyuken** [comment]
Thanks for getting this fixed so quickly. If this could be updated to ignore customizable sources (such as circadian lighting) this would be perfect. I am so happy I made the switch from CL to EC. This solves more issues than CL did for me, but in a perfect world I would have both of them able to work together. Now it seems like it is at least possible to make it happen!

**Comment by danobot** [comment]
You can add `ignored_event_sources` to your config. Note that this will only work if circadian lighting uses the new Context feature. You can enable ECs debug log to check the context id provided by circadian lighting.

**Comment by danobot** [comment]
Please test the new changes to ignore custom event sources and consider donating a few dollars to the project. Maintaining and implementing requests like these is very time consuming

**Comment by broyuken** [comment]
I will gladly test it once I find something that I can test against. And sure, I'll help maintain the project, it's incredible from the few weeks I've been using it.

**Comment by danobot** [comment]
Thanks I appreciate that, finding new contributors is difficult. For now just test the new config property and that adding additional ones does not break the fix. I don't want to delay this release as it might be months before other components are updates to use Context.

**Comment by broyuken** [comment]
I did have one of my lights block itself this afternoon, not sure how that happened. It was a smart bulb, not a switch. Doesn't happen every time though.

**Comment by danobot** [comment]
Did you add the `ignored_event_sources` to your EC instance? Something like the following should test the change:

```
ignored_event_sources:
    - anything1
    - anything2
```

EC will add itself to this list automatically.

**Comment by broyuken** [comment]
Does it need to be there if the only thing I want to ignore is EC? Because I haven't added that anywhere yet and it seems to be working fine most of the time.

**Comment by danobot** [comment]
no it doesn't need to be there, but we still have to test it before releasing it :)

**Comment by danobot** [comment]
Did you find any issues with this version and the `ignored_event_sources` configuration key?

**Comment by broyuken** [comment]
Sorry, I have not had a chance to test it fully yet. I just migrated from zwave2mqtt to qt-openzwave and have been dealing with some polling issues. I have seen a few times where a light gets blocked by itself, but I think it might be related to the polling. I may have also found another issue with sensor_resets_timer but I’ll probably open a new issue to track that.

**Comment by danobot** [comment]
I released this fix since someone else confirmed it worked, please create a new issue for other problems you encournter.
